src/frontend/src/lib/utils/transaction.utils.ts

```diff
@@ -29,3 +34,34 @@ export const mapTransactionIcon = ({
 					? IconSend
 					: IconReceive;
 };
+
+/**
+ * Group a list of transactions by date.
+ *
+ * It assumes that the transactions are already sorted by timestamp in descending order.
+ * It does not sort the transactions, nor the groups.
+ * If a transaction has no timestamp, it will be grouped under the key 'undefined'.
+ *
+ * @param transactions - List of transactions to group.
+ * @returns Object where the keys are the date and the values are the transactions for that date.
+ */
+export const groupTransactionsByDate = <
+	T extends BtcTransactionUi | EthTransactionUi | IcTransactionUi
+>(
+	transactions: T[]
+) => {
+	const currentDate = new Date();
+
+	return transactions.reduce<Record<string, T[]>>((acc, transaction) => {
+		if (isNullish(transaction.timestamp)) {
+			return { ...acc, undefined: [...(acc['undefined'] ?? []), transaction] };
+		}
+
+		const date = formatSecondsToNormalizedDate({
+			seconds: Number(transaction.timestamp),
+			currentDate
+		});
+
+		return { ...acc, [date]: [...(acc[date] ?? []), transaction] };
```

Comment:
the order of the transaction within the group does not matter?

Reply:
so, this function does not resort, it uses the same order that the transactions are provided in. So, if they are already sorted by date, for example, the order will be respected

Reply:
Maybe add a comment for our later selfs